File: python/session.py

from modelobjects import Cart, LocationServices


class Session:
    """Allows the OnlineShopping to access data classes
    and store them in the database"""
    def __init__(self):
        self.contents = {}
        self.put("CART", Cart())
        self.put("LOCATION_SERVICE", LocationServices())

    def put(self, key, value):
        self.contents[key] = value

    def get(self, key):
        return self.contents[key]

    def save_all(self):
        for key, value in self.contents.items():
            value.save_to_database()

    def __repr__(self):
        contents = ', '.join(f"{k}={v}" for k, v in self.contents.items())
        return f"""Session({contents})"""

File: typescript/src/Session.ts
import Cart from "./Cart";
import LocationService from "./LocationService";
import { IModelObject } from "./ModelObject";

/**
 * Allows the OnlineShopping to access data classes
 * and store them in the database.
 */
export default class Session {
    private readonly session: Map<string, IModelObject> = new Map();

    public Session() {
        this.session.set("CART", new Cart());
        this.session.set("LOCATION_SERVICE", new LocationService());
    }

    public get(key: string) {
        return this.session.get(key);
    }

    public put(key: string, value: IModelObject) {
        this.session.set(key, value);
    }

    public saveAll() {
        for (const entity of this.session.values()) {
            if (entity != null) {
                entity.saveToDatabase();
            }
        }
    }

    public toString() {
        let sessionContents = '\n';
        for (const [key, modelObject] of this.session) {
            sessionContents += key;
            sessionContents += "=";
            sessionContents += modelObject;
            sessionContents += "\n";
        }

        return "Session{" +
            sessionContents +
            "}";
    }
}
